Sources/MongoSwift/Compressor.swift

```diff
@@ -0,0 +1,32 @@
+/// Specifies a library to use for network compression.
+public struct Compressor {
+    /// Use zlib for data compression.
+    /// - SeeAlso: https://docs.mongodb.com/manual/reference/glossary/#term-zlib
+    public static let zlib = Compressor("zlib")
+
+    /// Use zlib for data compression, with the specified compression level.
+    /// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/#urioption.zlibCompressionLevel
+    public static func zlib(level: Int) throws -> Compressor {
+        try Compressor("zlib", level: level)
+    }
+
+    /// Compressor name.
+    internal let name: String
+
+    internal let zLibLevel: Int32?
```

Comment:
this `Compressor` struct could be used for more than just zlib, so we probably don't want a zlib specific value here. I think a protocol or something that encapsulates the behavior for attaching the compressor to the URI could be stored here instead.

Reply:
what I originally had here was an internal enum like
```swift
enum _Compressor {
    case zlib(level: Int? = nil)
    case zstd
}
```
I got rid of it because there was only one case, but I could bring it back now for the sake of making adding more compressors in the future easier.

Reply:
Yeah I think bringing this back would be nice.

Reply:
If we end up adding more properties to `zlib` we would just use a different case, e.g. `zlibV2(level: Int, comment: String)` ?

Reply:
switched to backing this struct with an internal enum.

since the enum is internal and we control all usages, we could just add a new property to the original zlib case. the main benefit is just that we get to be a little more strict about ensuring we don't have a level set for some future compressor that doesn't allow one.